### PR TITLE
fix bug when running image-to-text example

### PR DESCRIPTION
**Pull request by kaixuanliu**
change back value to 0 when do the padding. Before the fix, when we run `python3 run_pipeline.py  --model_name_or_path llava-hf/llava-1.5-7b-hf --use_hpu_graphs  --bf16 --use_kv_cache`, it will return wrong output:
`result = [[{'generated_text': 'USER:  \nWhat is shown in this image? ASSISTANT: The\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\n'}]
]`, this PR fix it.

### CODE REVIEW

File: optimum/habana/transformers/generation/utils.py [hunk 1]
```diff
@@ -957,7 +957,7 @@ def generate(
                             model_kwargs[other_inputs] = torch.nn.functional.pad(
                                 model_kwargs[other_inputs],
                                 (0, generation_config.max_new_tokens),
-                                value=generation_config.pad_token_id,
+                                value=0,
```
Comment (vidyasiv) [inline optimum/habana/transformers/generation/utils.py:960]: Could you explain what is the generation_config.pad_token_id and why it needs reset to 0?

**Comment by vidyasiv** [comment]
> change back value to 0 when do the padding. Before the fix, when we run `python3 run_pipeline.py --model_name_or_path llava-hf/llava-1.5-7b-hf --use_hpu_graphs --bf16 --use_kv_cache`, it will return wrong output: `result = [[{'generated_text': 'USER: \nWhat is shown in this image? ASSISTANT: The\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\n'}] ]`, this PR fix it.

Can you also paste the output after the fix?

**Comment by vidyasiv** [comment]
@jiminha let me know if you are able to review further.

**Comment by vidyasiv** [comment]
Fix incorporated into https://github.com/huggingface/optimum-habana/pull/1366, please review there

**Comment by kaixuanliu** [comment]
> > change back value to 0 when do the padding. Before the fix, when we run `python3 run_pipeline.py --model_name_or_path llava-hf/llava-1.5-7b-hf --use_hpu_graphs --bf16 --use_kv_cache`, it will return wrong output: `result = [[{'generated_text': 'USER: \nWhat is shown in this image? ASSISTANT: The\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\nЪ\n'}] ]`, this PR fix it.
> 
> Can you also paste the output after the fix?

after the fix , the output is:
`result = [[{'generated_text': 'USER:  \nWhat is shown in this image? ASSISTANT: The image shows a pier extending out over a body of water, with a mountain in the background. The pier is made of wood and appears to be a popular spot for people to gather and enjoy the view.'}]]`

**Review by jiminha — APPROVED** [review]

File: optimum/habana/transformers/generation/utils.py (hunk 1 above)
Comment (jiminha) [reply optimum/habana/transformers/generation/utils.py:960]: ["attention_mask", "token_type_ids" -> this should be set to 0.

Comment (kaixuanliu) [reply optimum/habana/transformers/generation/utils.py:960]: The `generation_config.pad_token_id` is 32001. And the original value for padding here is `0` before this PR: [L963](https://github.com/huggingface/optimum-habana/commit/d28bae9277b1078512a60a9f3c793aede2c873fc#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eL963).

**Comment by zongwave** [comment]
> @jiminha let me know if you are able to review further. @zongwave can you review this as it changes back the code you had added for input embeddings

In this context, it intend to pad the "attention_mask," so it should indeed be set to 0.

**Review by zongwave — APPROVED** [review]

**Review by regisss — APPROVED** [review]
Good catch!
